src/gs-plugin-loader.c

```diff
@@ -3909,6 +3959,21 @@ gs_plugin_loader_class_init (GsPluginLoaderClass *klass)
 }
 
 static void
+gs_plugin_loader_settings_changed_cb (GSettings *settings,
+				      const gchar *key,
+				      GsPluginLoader *plugin_loader)
+{
+	GsPluginLoaderPrivate *priv = gs_plugin_loader_get_instance_private (plugin_loader);
+	if (g_strcmp0 (key, "allow-updates") == 0) {
+		priv->allow_updates = g_settings_get_boolean (priv->settings,
```

Comment:
There's really no need to store this in a private field. Calling `g_settings_get()` is fast, and you can do it directly from the property getter.